base_layer/p2p/src/config.rs

```diff
@@ -131,9 +129,8 @@ impl Default for P2pConfig {
             transport: Default::default(),
             datastore_path: PathBuf::from("peer_db"),
             peer_database_name: "peers".to_string(),
-            max_concurrent_inbound_tasks: 50,
-            max_concurrent_outbound_tasks: 100,
-            outbound_buffer_size: 100,
+            max_concurrent_inbound_tasks: 4,
+            max_concurrent_outbound_tasks: 4,
```

Comment:
With the latency we seem to be getting on tor sometimes, is this perhaps not a bit too low by default?

Reply:
This is the number of concurrent inbound/outbound tasks that the pipeline (i.e. DHT middleware) runs.
Tor latency is what it is. These two parts of your question don't really have anything to do with each other.
If anything, higher latency = slower message rate = concurrency is used less. 

The logs indicate that the limits of 4/4 are not hit that often, even in stress test conditions, and usage is cleared extremely quickly. One main main reason for reducing the concurrency is that if something locks up, the limit is hit quickly instead of hiding the issue for extended periods of time, possibly leading to us never discovering them at all. Another reason is to limit the busy-ness of the node under stress conditions, which under high loads, leads to peer backpressure (sender has to reduce the rate that it sends messages)

Reply:
We may have to increase concurrency, but I propose that we keep it low and increase if there is an issue. The limits that were there were way too high (should never really be higher than the number of cores) and led to the problems mentioned as well as too much buffering of messages.

Reply:
sounds good. Thanks